semgrep_output_v1.atd

```diff
@@ -165,11 +165,38 @@ type error_severity
   | Info <json name="info">
 ]
 
-(* TODO? PRO can actually have "flavors" (see Engine_type.ml) *)
+(* Used for a best-effort report to users about what findings they get with
+   the pro engine that they couldn't with the oss engine.
+
+   Intrafile_taint = requires intrafile taint
+   Interfile_taint = requires interfile taint
+   Other_pro_feature = requires some non-taint pro feature *)
+type pro_feature
+  <ocaml attr="deriving show">
+  <python decorator="dataclass(frozen=True)"> = [
```

Comment:
Should this be a record with Booleans? May it be easier to extend in the future ? Not a strong opinion, but I'm thinking what happens when we want to track features A and B and a finding can involve both.

Reply:
I think you're right. I don't love the idea of this being a permanent system for reporting a matrix of pro features used to customers, but it doesn't hurt to be flexible.